platform/wayland/cog-platform-wl.c

```diff
@@ -1819,6 +1651,30 @@ cog_wl_platform_create_im_context(CogPlatform *platform)
     return NULL;
 }
 
+void
+cog_wl_platform_popup_create(CogWlPlatform *platform, WebKitOptionMenu *option_menu)
+{
+    g_assert(!platform->popup);
+
+    CogWlPopup *popup = cog_wl_popup_create(platform, option_menu);
+    platform->popup = popup;
+}
+
+void
+cog_wl_platform_popup_destroy(CogWlPlatform *platform)
+{
+    g_assert(platform->popup);
+    cog_wl_popup_destroy(platform->popup);
+    platform->popup = NULL;
```

Comment:
The `g_clear_pointer()` macro can be used here:

```suggestion
    g_clear_pointer(&platform->popup, cog_wl_popup_destroy);
```